### PR TITLE
[Staging & Production] Add-on Configuring Confirmation confirm

### PR DESCRIPTION
**Pull request by haoyuchen1992**
### Purpose
In our project and profile add-on setting pages, only when users click submit button, will the changes be saved. In case that users forget to submit, a confirmation page should be added when detecting there are unsubmitted add-on setting changes. 
Issues resolve -- #2155 

### Changes
The basic idea is to bind [beforeunload event](http://www.w3schools.com/jsref/event_onbeforeunload.asp). Before unloading, detect whether there are changed settings. If yes, return a static warning message.
screen shot 2015-06-17 at 4 54 55 pm

### Side effects
In our current style guide, it doesn't define the modal panel for confirmation.  So it doesn't follow the guideline for general modal panel.

### CODE REVIEW

File: website/static/js/pages/profile-settings-addons-page.js [hunk 1]
```diff
@@ -89,6 +91,18 @@ for (var i=0; i < addonEnabledSettings.length; i++) {
    }
 }
 
+/* Before closing the page, Check whether the newly checked addon are updated or not */
+$(window).on('beforeunload',function() {
+    //new checked items but not updated
+    var checked = uncheckedOnLoad.filter($('#selectAddonsForm input:checked'));
```
Comment (lyndsysimon) [inline website/static/js/pages/profile-settings-addons-page.js]: jQuery's `.filter()` should allow you to pass in a selector directly, making the `$()` unnecessary.

```javascript
var checked = uncheckedOnLoad.filter($('#selectAddonsForm input:checked'));
```

**Comment by lyndsysimon** [comment]
Looks great - first pass complete :pager:

@sloria I'm find with this being merged as-is if you get to it before Haoyu responds to my comment.

File: website/static/js/pages/profile-settings-addons-page.js (hunk 1 above)
Comment (haoyuchen1992) [inline website/static/js/pages/profile-settings-addons-page.js]: That's true! It is not necessary. I have changed it. But I just found this fix had a logic problem, which would invoke the unnecessary warming message when submitting. I will keep working on it!

Comment (haoyuchen1992) [inline website/static/js/pages/profile-settings-addons-page.js]: @lyndsysimon  I just updated the input for the filter method and resolved the unexpected warning message when submitting. Thanks!

**Pull request merged**